core/build.gradle

```
@@ -31,7 +31,7 @@ spotless {
 }
 
 def versionInfo = tasks.register("micronautVersionInfo", WriteProperties) {
-    outputFile = "${buildDir}/resources/version/micronaut-version.properties"
+    destinationFile = file("${layout.buildDirectory.asFile.get()}/resources/version/micronaut-version.properties")
```

:x: `destinationFile` is a proper `RegularFileProperty`

```suggestion
    destinationFile = layout.buildDirectory.file("resources/version/micronaut-version.properties")
```

---

fixed